Vamsidhar taking over for Samman: Backend: Phase 2 Summary Dashboard: Making the Issues for Injuries Tracking

SammanBaidya opened the pull request:
# Description
![9](https://github.com/user-attachments/assets/ee56c122-e4eb-4620-8cea-6c11fcb7f55d)


## Related PRS (if any):
This backend PR is related to the #5084 frontend PR. [https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/5084]
To test this backend PR you need to checkout the #5084 frontend PR. https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/5084
…

## Main changes explained:
- Created API to create Injury Issue, delete, rename, copy and get injury issue.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally by npm run start build then npm start.
3. Clear site data/cache
4. log as admin user
5. verify function “A” (feel free to include screenshot here)

## Test for these requirements:

![10](https://github.com/user-attachments/assets/b86ca1b3-5331-492f-bcc0-6a5857b5c522)
![11](https://github.com/user-attachments/assets/2f063a0f-ad46-41f5-add7-5ff9077e8832)


## Screenshots or videos of changes:
[test run.webm](https://github.com/user-attachments/assets/f688c54e-1ea2-495d-b2ec-cfed6851be59)

SammanBaidya committed:
feat: add rename, copy, delete, post api for injury Issue

SammanBaidya committed:
feat: add Jest mocks

rahul07bagul reviewed (approved):
Working as expected, tested with frontend PR 3544. Looks good.

knsspsj19 commented:
Could be helpful if we preprocess the data for clarity:
Make the column names descriptive. with appropriate Datatypes. Eg: the "Open Since" column is text field with values like "3D" whereas it could be a timestamp in the "hh:mm:ss" format.
Additional comments in the corresponding frontend PR.

TaariqMansurie reviewed (approved):
Everything is working as expected. Tested in conjunction with frontend PR #3544.

SammanBaidya committed:
merge conflict fix

SammanBaidya committed:
merge conflict fix

SammanBaidya committed:
Merge branch 'development' into Samman-injurytracking

SammanBaidya committed:
merge conflict fix

vamsidharpanithi committed:
Merge branch 'development' into Samman-injurytracking

vamsidharpanithi committed:
fix: resolve undefined dates variable and linting issues

vamsidharpanithi committed:
chore: merge origin/development into Samman-injurytracking

Resolve conflicts: BM injury/building issues, routes, automation services,
email flows, dependencies, and Node 20 in .nvmrc.

vamsidharpanithi committed:
fix: prevent mass assignment in BM issue create endpoints

vamsidharpanithi committed:
fix: prevent mass assignment in BM issue create endpoints

Anusha-Gali reviewed (approved):
Hi Vamsidhar,

I have reviewed your PR locally and it works as per requirement in context of API's.

<img width="1263" height="760" alt="Image" src="https://github.com/user-attachments/assets/6c0a9cbe-19b8-4821-81a8-322f2aa6ac84" />
<img width="1263" height="760" alt="Image" src="https://github.com/user-attachments/assets/355e961b-92b8-4f42-b512-6a34cabe3baa" />
<img width="1263" height="760" alt="Image" src="https://github.com/user-attachments/assets/71da73cc-7428-4cf5-931a-e8bea6acc06b" />

one-community commented:
Thank you all, merging!
